overlord/snapshotstate/backend/backend.go

```diff
@@ -330,7 +337,8 @@ func Save(ctx context.Context, id uint64, si *snap.Info, cfg map[string]interfac
 
 	w := zip.NewWriter(aw)
 	defer w.Close() // note this does not close the file descriptor (that's done by hand on the atomic writer, above)
-	if err := addDirToZip(ctx, snapshot, w, "root", archiveName, si.DataDir()); err != nil {
+	savingUserData := false
```

Comment:
I considered adding it, but I think it would be improper: this option is "dynamic", we'll change it across `addDirToZip()` executions in the same way as we change the username, archive name or data dir; whereas the addDirToZipOptions contains fixed options.

Reply:
I see the reasoning but I wonder if we should instead just move more things to archiveOptions. The current signature of addDirToZip is getting a bit unwieldy. Plus I'm not a fan of booleans in function signatures. Here it's elegantly worked around though.

Reply:
Fwiw, the original comemnt about `addDirToZipOptions` is no longer needed, I removed the options and just include the `[]excludeList` in the function now. I got a strong YAGNI and we can always add back the options if we have more than a single entry in them.